test/hotspot/jtreg/runtime/Monitor/libIterateMonitorWithDeadObjectTest.c

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright (c) 2022, 2023, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+#include <jni.h>
+#include <stdlib.h>
+#include <pthread.h>
+#include <stdio.h>
+#include <unistd.h>
+
+#define die(x) do { printf("%s:%s\n",x , __func__); perror(x); exit(EXIT_FAILURE); } while (0)
+
+#ifndef _Included_IterateMonitorWithDeadObjectTest
+#define _Included_IterateMonitorWithDeadObjectTest
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+static JavaVM* jvm;
+static pthread_t attacher;
+
+static jobject create_object(JNIEnv* env) {
+  jclass clazz = (*env)->FindClass(env, "java/lang/Object");
+  if (clazz == 0) die("No class");
```

Comment:
The `die` method is for errors with system calls. It won't show useful information for JNI calls that leave exceptions pending.

Reply:
I've added better checks.